File: longestSequence/longestSequence.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        numSet = set(nums) # Convert into a hash set for an O(1) lookup
        longest = 0

        for num in numSet:
            if (num - 1) not in numSet: # Start a new sequence
                length = 1
                while (num + length) in numSet:
                    length += 1
                longest = max(length, longest)
        return longest

File: longestSequence/longestSequence.cpp
#include <vector>
#include <unordered_set>

class Solution {
public:
    int longestConsecutive(std::vector<int>& nums) {
        std::unordered_set<int> hashSet;

        for (int num : nums){
            hashSet.insert(num);
        }

        int maxLen = 0;

        for (int num : hashSet){
            int curLen = 0;
            if (hashSet.find(num-1) == hashSet.end()){
                curLen++;

                while (hashSet.find(num+curLen) != hashSet.end()){
                    curLen++;
                }

                if (curLen > maxLen){
                    maxLen = curLen;
                }
                
            }
        }

        return maxLen;

    }
};